src/frameworks/next/index.ts

```diff
@@ -344,25 +344,23 @@ export async function ɵcodegenFunctionsDirectory(sourceDir: string, destDir: st
   const { distDir } = await getConfig(sourceDir);
   const packageJson = await readJSON(join(sourceDir, "package.json"));
   if (existsSync(join(sourceDir, "next.config.js"))) {
-    let esbuild;
-    try {
-      esbuild = await import("esbuild");
-    } catch (e: unknown) {
-      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
-      logger.debug(`Failed to load 'esbuild': ${e}`);
-      throw new FirebaseError(
-        `Unable to find 'esbuild'. Install it into your local dev dependencies with 'npm i --save-dev esbuild''`
-      );
-    }
-    await esbuild.build({
-      bundle: true,
-      external: Object.keys(packageJson.dependencies),
-      absWorkingDir: sourceDir,
-      entryPoints: ["next.config.js"],
-      outfile: join(destDir, "next.config.js"),
-      target: `node${NODE_VERSION}`,
-      platform: "node",
-    });
+    // Bundle their next.config.js with esbuild via NPX, pinned version was having troubles
+    // on m1 macs, older Node versions, and eitherway we should avoid taking on any deps in firebase-tools
```

Comment:
`on m1 mac, older Node version, and eitherway` => `on m1 macs and older Node versions; either way`